src/main/java/org/openrewrite/java/migrate/AddMissingMethodImplementation.java

```diff
@@ -0,0 +1,148 @@
+        this.fullyQualifiedClassName = fullyQualifiedClassName;
+        this.methodPattern = methodPattern;
+        this.methodTemplateString = methodTemplateString;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Adds missing method implementations.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Check for missing methods required by interfaces and adds them.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>(fullyQualifiedClassName, true), new ClassImplementationVisitor());
+    }
+
+    public class ClassImplementationVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final JavaTemplate methodTemplate = JavaTemplate.builder(methodTemplateString).build();
+        private final MethodMatcher methodMatcher = new MethodMatcher(methodPattern, true);
+
+        public boolean matchesInterface(JavaType.Class type) {
+            if (type != null) {
+                if (type.getFullyQualifiedName().equals(fullyQualifiedClassName)) {
+                    return true;
+                }
+
+                // check for matches on super interface
+                List<JavaType.FullyQualified> superInterfaces = type.getInterfaces();
+                boolean foundOnSuperInterface = false;
+                for (JavaType.FullyQualified superInterface : superInterfaces) {
+                    if (matchesInterface((JavaType.Class) superInterface)) {
+                        foundOnSuperInterface = true;
+                        break;
+                    }
+                }
+                return foundOnSuperInterface;
+            }
+
+            return false;
+        }
+
+        public boolean implementsInterface(J.ClassDeclaration classDecl) {
+            List<TypeTree> implementedClasses = classDecl.getImplements();
+            if (implementedClasses != null) {
+                for (TypeTree implementedClass : implementedClasses) {
+                    JavaType.Class type = (JavaType.Class) implementedClass.getType();
+                    return matchesInterface(type);
+                }
+            }
+
+            // also check extended class to handle abstract classes
+            TypeTree extendedClass = classDecl.getExtends();
+            if (extendedClass != null) {
+                JavaType.Class type = (JavaType.Class) extendedClass.getType();
+                return matchesInterface(type);
+            }
+            return false;
+        }
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext executionContext) {
+            boolean implementsInterface = implementsInterface(classDecl);
+            // Don't make changes to classes that don't match the fully qualified name
+            if (classDecl.getType() == null || !implementsInterface) {
+                return classDecl;
```

Comment:
Shouldn't we call super here just to be sure we visit any sub classes too?

Reply:
That's interesting, so calling super here would take care of subclasses as well. In the other PR I put out yesterday, I ended up manually searching for subclassed. Additional info about this would be good to have.
So I would just change line 128 to `super.visitClassDeclaration(classDecl, executionContext)`?

Reply:
Usually calling the super method is the first thing we do in an overridden method, and exclusively use the result from that throughout the method after that. Here's a part of the docs that explains that in some more detail:
https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices#be-deliberate-about-lst-traversal

Reply:
Thanks, Tim, that link was very helpful. I've gone ahead and made the necessary updates in both PRs. I also included a new test in this PR to confirm that sub-classes are detected and updated.